server-client/index.js

```diff
@@ -80,10 +80,13 @@ class ServerClient {
     if (this.clientId) {
       this.app.clientIds.delete(this.clientId)
       this.app.nodeIds.delete(this.nodeId)
-      for (let i in this.app.subscribers) {
-        delete this.app.subscribers[i][this.nodeId]
-        if (Object.keys(this.app.subscribers[i]).length === 0) {
-          delete this.app.subscribers[i]
+      let lastDestroyIndex = 0
+      for (let channel in this.app.subscribers) {
+        if (this.app.subscribers[channel][this.nodeId]) {
+          this.app.log.add(
+            { type: 'logux/unsubscribe', channel },
+            { id: `destroy${++lastDestroyIndex} ${this.nodeId}` }
```

Comment:
It is not valid [Logux ID](https://logux.io/guide/concepts/meta/#id-and-time).

It is better to avoid generating ID from another node since it could be not unique.

Reply:
It means that logic will be more complicated. Seems like we will need to call the `unsubscribe` callback manually in the `disconnect` callback.

Reply:
The purpose of faking the id is to make this action be processed as if it was emitted by a client (nodeId is extracted from id right now).

What would you suggest to solve this? Replacing `destroy{N}` with server local time?

Reply:
So your suggestion is to get rid of emitting 'logux/unsubscribe' in case of disconnect and duplicate this logic in ServerClient?

Reply:
> So your suggestion is to get rid of emitting 'logux/unsubscribe' in case of disconnect

You may want to create `logux/unsubscribe` action to pass it to `unsubscribe` handler.

> duplicate this logic in ServerClient

Yeap. But of course you can avoid duplication by moving this logic to some private method in `BaseServer`.

Reply:
Done

Reply:
Looks great. Just fix the test and we are ready for merge.

Reply:
Done :)

Reply:
btw is that only my local macOS setup issue or `reports EADDRINUSE error` test snapshot is outdated?
```
  ● reports EADDRINUSE error

    expect(received).toMatchSnapshot()

    Snapshot name: `reports EADDRINUSE error 1`

    - Snapshot  - 1
    + Received  + 1

    - {"level":60,"time":"1970-01-01T00:00:00.000Z","pid":21384,"hostname":"localhost","name":"test","err":{},"note":"Another Logux server or other app already running on this port. Probably you haven’t stopped server from other project or previous version of this server was not killed.\n\n$ su - root\n# netstat -nlp | grep 31337\nProto   Local Address   State    PID/Program name\ntcp     0.0.0.0:31337    LISTEN   `777`/node\n# sudo kill -9 `777`","msg":"Port `31337` already in use"}
    + {"level":60,"time":"1970-01-01T00:00:00.000Z","pid":21384,"hostname":"localhost","name":"test","err":{},"note":"Another Logux server or other app already running on this port. Probably you haven’t stopped server from other project or previous version of this server was not killed.\n\n$ sudo lsof -i 31337\n$ sudo kill -9 `<processid>`","msg":"Port `31337` already in use"}
      ↵
```

```
  ● reports EADDRINUSE error

    expect(received).toMatchSnapshot()

    Snapshot name: `reports EADDRINUSE error 2`

    - Snapshot  - 5
    + Received  + 2

       FATAL  Port 31337 already in use at 1970-01-01 00:00:00
              Another Logux server or other app already running on this port.
              Probably you haven’t stopped server from other project or previous
              version of this server was not killed.

    -         $ su - root
    -         # netstat -nlp | grep 31337
    -         Proto   Local Address   State    PID/Program name
    -         tcp     0.0.0.0:31337    LISTEN   777/node
    +         $ sudo lsof -i 31337
    -         # sudo kill -9 777
    +         $ sudo kill -9 <processid>
```

Reply:
(that is my lame reason of missing failing test in plethora of other failing test output:)